plugins/webp-uploads/hooks.php

```diff
@@ -747,13 +747,84 @@ function webp_uploads_render_generator(): void {
 }
 add_action( 'wp_head', 'webp_uploads_render_generator' );
 
+/**
+ * Process a block's content to handle background images for specific block types.
+ *
+ * This function targets blocks like Cover and Group that may use background images,
+ * converting them to modern image formats when appropriate.
+ *
+ * @since n.e.x.t
+ *
+ * @param string               $block_content The block content.
+ * @param array<string, mixed> $block         The block.
+ * @return string The filtered block content.
+ */
+function webp_uploads_filter_block_background_images( string $block_content, array $block ): string {
+	// Only run on frontend.
+	if ( ! webp_uploads_in_frontend_body() || '' === $block_content ) {
+		return $block_content;
+	}
+
+	$attachment_id = null;
+	$image_url     = '';
+
+	// Process Cover block.
+	if ( 'core/cover' === $block['blockName'] && isset( $block['attrs']['id'] ) ) {
+		$attachment_id = $block['attrs']['id'];
+		$image_url     = isset( $block['attrs']['url'] ) ? $block['attrs']['url'] : '';
+	}
+
+	// Process Group block with background image.
+	if ( 'core/group' === $block['blockName'] && isset( $block['attrs']['style']['background']['backgroundImage'] ) ) {
+		$bg_image = $block['attrs']['style']['background']['backgroundImage'];
+
+		if ( isset( $bg_image['id'] ) ) {
+			$attachment_id = $bg_image['id'];
+			$image_url     = isset( $bg_image['url'] ) ? $bg_image['url'] : '';
+		}
+	}
+
+	// If we have a valid attachment and URL, process it.
+	if ( ! is_null( $attachment_id ) && '' !== $image_url ) {
+		$metadata = wp_get_attachment_metadata( $attachment_id );
+
+		if ( is_array( $metadata ) ) {
+			$original_mime = get_post_mime_type( $attachment_id );
+
+			if ( null !== $original_mime && '' !== $original_mime ) {
+				$target_mimes = webp_uploads_get_content_image_mimes( $attachment_id, 'background_image' );
+
+				foreach ( $target_mimes as $target_mime ) {
+					if ( $target_mime === $original_mime ) {
+						continue;
+					}
+
+					$new_url = webp_uploads_get_mime_type_image( $attachment_id, $image_url, $target_mime );
+
+					if ( null !== $new_url && '' !== $new_url ) {
+						// Replace the URL in the content.
+						$block_content = str_replace( $image_url, $new_url, $block_content );
```

Comment:
I think it would be better to Would this be better to use HTML Tag Processor to do the replacement specifically in the context of the `style` attribute. It could even append the new style property as opposed to doing a replacement, since the later CSS property will override the former. (Maybe a string replacement inside the attribute is still better though.)

Also, shouldn't this `break` the `foreach` loop when a replacement is done?

```suggestion
						$processor = new WP_HTML_Tag_Processor( $block_content );
						while ( $processor->next_tag() ) {
							$style = $processor->get_attribute( 'style' );
							if ( is_string( $style ) && str_contains( 'background-image:', $style ) && str_contains( $image_url, $style ) ) {
								$style .= sprintf( 'background-image: url( "%s" );', $new_url );
								$processor->set_attribute( 'style', $style );
								$block_content = $processor->get_updated_html();
								break 2;
							}
						}
```